app/Commands/Extra/Compile.hs

```diff
@@ -172,14 +172,19 @@ native64Args buildDir o outfile inputFile =
     <> [ "-DARCH_NATIVE64",
          "-DAPI_LIBC",
          "-m64",
-         "-O3",
+         "-O" <> show optValue,
          toFilePath inputFile
        ]
     <> ( if
              | not (o ^. compilePreprocess || o ^. compileAssembly) ->
                  ["-ljuvix"]
              | otherwise -> []
        )
+  where
+    optValue :: Int
+    optValue
+      | o ^. compileDebug = 1
```

Comment:
If somebody has overridden the optimization level by explicitly passing `-On` then shouldn't it be taken into account?